### PR TITLE
PP-9236: Run provider contract tests

### PR DESCRIPTION
**Pull request by oswaldquek**
Successful provider contract test build: https://github.com/alphagov/pay-adminusers/runs/5558155471?check_suite_focus=true

### CODE REVIEW

File: .github/workflows/post-merge.yml [hunk 1]
```diff
@@ -1,6 +1,7 @@
 name: Post Merge
 
 on:
+  pull_request: #to be delete
```
Comment (oswaldquek) [inline .github/workflows/post-merge.yml:4]: Will delete when PR is approved

File: .github/workflows/run-tests.yml [hunk 2]
```diff
@@ -2,7 +2,7 @@ name: Github Actions Tests
 
 on:
   workflow_call:
-  pull_request:
+#  pull_request:
```
Comment (oswaldquek) [inline .github/workflows/run-tests.yml:5]: Will revert when PR is approved

**Review by jfharden — COMMENTED** [review]
Great 👍 

I'm happy to approve this. Can you get it into the "final state". (I know post-merge wont run then but the unit and integration tests will which we currently have required and aren't running on this pr. It's demonstrated that the post-merge is good 👍 )

File: .github/workflows/run-tests.yml [hunk 3]
```diff
@@ -29,4 +29,6 @@ jobs:
         run: |
           docker pull govukpay/postgres:11.1
       - name: Run integration tests
-        run: mvn clean verify
+        run: |
+          echo "${{ github.sha }}"
```
Comment (jfharden) [inline .github/workflows/run-tests.yml:33]: Don't think you need this anymore?

File: .github/workflows/post-merge.yml [hunk 4]
```diff
@@ -1,6 +1,7 @@
 name: Post Merge
 
 on:
+  pull_request:
```
Comment (jfharden) [inline .github/workflows/post-merge.yml:4]: This will be removed right?

**Review by jfharden — COMMENTED** [review]
I approved the other PR this depends on but I think you've left a couple of bits in here

File: .github/workflows/post-merge.yml (hunk 4 above)
Comment (oswaldquek) [reply .github/workflows/post-merge.yml:4]: oh I only deleted the `#to be deleted` comment 😆

Comment (oswaldquek) [reply .github/workflows/post-merge.yml]: fixed

File: .github/workflows/run-tests.yml (hunk 3 above)
Comment (oswaldquek) [reply .github/workflows/run-tests.yml]: Fixed

File: .github/workflows/post-merge.yml (hunk 4 above)
Comment (jfharden) [reply .github/workflows/post-merge.yml]: Classic 😄

**Review by jfharden — APPROVED** [review]

**Pull request merged**